### DISCUSSION TITLE
429 Client error

### DISCUSSION THREAD

**Discussion by RayHJLin** [matin/garth#60] [Q&A] [answered]
Hi,

As title. I always get this error during garth.login

Does anyone has suggestion?

Thanks.

**Comment by matin** [comment, marked as answer]
429 is for too many requests. Do you have anyway to try from a different IP address?

**Reply by RayHJLin** [reply]
I changed to my mobile network and it works. Thanks but why always return 429 error when I used my wifi?

**Reply by matin** [reply]
Your internet likely shares a IP address with others, and that IP address is blocked. At least, that's my theory.

**Comment by weberk** [comment]
@matin : Some questions on this situation I also encountered:
- Are there any insights on how long an IP remains blocked due to an HTTP 429 error? 
- If additional login attempts are made during the block, does it extend the timer? 
- Does the HTTP 429 error response contain any information about the duration of the block? (i.e. Header Fields Retry-After/RateLimit-Reset)

**Reply by matin** [reply]
Are you saving and reusing the login tokens?

From what I've seen, the rate limit is based primarily on login, but that can be addressed by saving and reusing the auth tokens.

**Reply by weberk** [reply]
No, I was not saving when testing - will do. 

Next day login was working again. Seems the source IP is not blocked forever.

**Reply by LukeRabauke** [reply]
i have the same problem since yesterday.
so i guess i will deactivate my automation and wait 1-2 days - hmmmm

**Reply by weberk** [reply]
Yes, same here. I waited already 12h, but the next access was again blocked by rate limit http error code 429.

I am reusing the auth1 token, as well as auth2 token. But since garth is seeing auth2 token expired it uses auth1 token to get a fresh auth2 token. Here the 429 is thrown. 

@matin : is it necessary to attach a auth2 token to requests, or does it also work with auth1 tokens only?
@LukeRabauke : are you having your automation running on eu.pythonanywhere.com? At least my deployment runs there and maybe some other too, sharing the same/few outgoing IPs

**Reply by LukeRabauke** [reply]
@weberk no I am Running local in my Raspi HomeAssistant Setup or for Development on my MacBook. Both not working recently without changes in my Code :(

**Reply by weberk** [reply]
@LukeRabauke : your whole LAN (macBook, Raspi) is using the same outgoing IP and therfore you haven been banned because making to many auth requests/time interval.

**Reply by mikegithubwaw** [reply]
I have the same error since last couple of days, waited 24h no luck

**Reply by alexsanjoseph** [reply]
I'm having the same issue in my Production Railway instances - On local it is working fine for now (but I assume it will break soon)

**Reply by alexsanjoseph** [reply]
Primary discussion happening here: https://github.com/matin/garth/issues/217
